packages/odyssey-react-mui/src/components/TextField/TextField.tsx

```diff
@@ -0,0 +1,207 @@
+   */
+  autoComplete?: string;
+  /**
+   * If `true`, the component is disabled.
+   */
+  disabled?: boolean;
+  /**
+   * End `InputAdornment` for this component.
+   */
+  endAdornment?: React.ReactNode;
+  /**
+   * If `error` is not undefined, the `input` will indicate an error.
+   */
+  error?: string;
+  /**
+   * The helper text content.
+   */
+  hint?: string;
+  /**
+   * The id of the `input` element.
+   */
+  id?: string;
+  /**
+   * The label for the `input` element.
+   */
+  label?: string;
+  /**
+   * If `true`, a [TextareaAutosize](/material-ui/react-textarea-autosize/) element is rendered.
+   */
+  multiline?: boolean;
+  /**
+   * Callback fired when the value is changed.
+   */
+  onChange?: React.ChangeEventHandler<HTMLTextAreaElement | HTMLInputElement>;
+  /**
+   * Callback fired when the `input` element get focus.
+   */
+  onFocus?: React.FocusEventHandler<HTMLInputElement | HTMLTextAreaElement>;
+  /**
+   * The label for the `input` element if the it's not optional
+   */
+  optionalLabel?: string;
+  /**
+   * The short hint displayed in the `input` before the user enters a value.
+   */
+  placeholder?: string;
+  /**
+   * It prevents the user from changing the value of the field
+   */
+  readOnly?: boolean;
+  /**
+   * If `true`, the `input` element is required.
+   */
+  required?: boolean;
+  /**
+   * Start `InputAdornment` for this component.
+   */
+  startAdornment?: React.ReactNode;
+  /**
+   * Type of the `input` element. It should be [a valid HTML5 input type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Form_%3Cinput%3E_types).
+   */
+  type?: "email" | "password" | "search" | "tel" | "text";
+  /**
+   * The value of the `input` element, required for a controlled component.
+   */
+  value?: unknown;
+}
+
+export const TextField = forwardRef<HTMLInputElement, TextFieldProps>(
+  (props, ref) => {
+    const {
+      autoComplete,
+      disabled = false,
+      endAdornment,
+      error,
+      hint,
+      id: idOverride,
+      label,
+      multiline = false,
+      onChange,
+      onFocus,
+      optionalLabel,
+      placeholder,
+      readOnly,
+      required = true,
+      startAdornment,
+      type: inType = "text",
+      value,
+    } = props;
+
+    const [inputType, setInputType] = useState(inType);
+    const handleClickShowPassword = () => {
+      setInputType((currentType) =>
+        currentType === "password" ? "text" : "password"
+      );
+    };
+    const handleMouseDownPassword = (
+      event: React.MouseEvent<HTMLButtonElement>
+    ) => {
+      event.preventDefault();
+    };
+
+    const id = useUniqueId(idOverride);
+    const hintId = hint && id ? `${id}-hint` : undefined;
+    const errorId = error && id ? `${id}-error` : undefined;
+    const labelId = label && id ? `${id}-label` : undefined;
+
+    return (
+      <FormControl disabled={disabled} error={!!error}>
```

Comment:
```suggestion
      <FormControl disabled={disabled} error={Boolean(error)}>
```

I would also recommend we rename the `error` prop to `errorMessage`.

Reply:
+1 to `errorMessage`